storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -139,12 +139,12 @@
                   (.add local pair) 
 
                   ;;Using java objects directly to avoid performance issues in java code
-                  (let [node+port (get @task->node+port task)]
-                    (when (not (.get remoteMap node+port))
-                      (.put remoteMap node+port (ArrayList.)))
-                    (let [remote (.get remoteMap node+port)]
+                  (let []
+                    (when (not (.get remoteMap task))
+                      (.put remoteMap task (ArrayList.)))
+                    (let [remote (.get remoteMap task)]
                       (.add remote (TaskMessage. task (.serialize serializer tuple)))
-                     )))) 
+                    ))))
```

Comment:
@d2r 
Maybe I misunderstood your comment.
Seems like I grouped tuples by task, not node+port. It should be grouped again from mk-transfer-tuples-handler.
I'll fix it. Maybe TransferDrainer can help this, so please forget about last comment.